sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/basicLogicalOperators.scala

```diff
@@ -924,11 +947,6 @@ case class SubqueryAlias(
     child.output.map(_.withQualifier(qualifierList))
   }
 
-  override def metadataOutput: Seq[Attribute] = {
-    val qualifierList = identifier.qualifier :+ alias
-    child.metadataOutput.map(_.withQualifier(qualifierList))
-  }
```

Comment:
Why is the logic here removed? Won't this cause resolution failures when referencing a metadata column via an alias? Like `SELECT s._file FROM (SELECT ...) s`?

Reply:
@cloud-fan, should we support this case if it requires changing the query during analysis after being resolved?

Reply:
I think we should only expose the metadata column in a single SELECT group, e.g. `SELECT _file FROM t`, `SELECT t1._file FROM t1 JOIN t2`.

It's super weird if we can propagate the metadata column through SELECT groups, e.g. `SELECT s._file FROM (SELECT a, b FROM t) s`. The `s` is a subquery alias and the subquery has a clear output list which is `a, b`. It may surprise users if they can access `s._file`.

However, I do agree with @rdblue that simple alias should be supported. For example, `SELECT t1._file FROM t t1 JOIN t t2`. `SELECT s._file FROM (SELECT ...) s` won't work anyway because `Project` can't propagate the metadata columns.

That said, let's propagate metadata columns in `SubqueryAlias`